tests/serialization_tests/tests/signing_ser.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+// Doesn't run these unless feature specified
+#![cfg(feature = "submodule_tests")]
+#![allow(dead_code, non_snake_case)]
+
+use bls_signatures::{PrivateKey, Serialize};
+use crypto::{signature, Signature};
+use encoding::Cbor;
+use forest_message::{unsigned_message, UnsignedMessage};
+use serde::Deserialize;
+use std::fs::File;
+use std::io::prelude::*;
+
+#[derive(Deserialize)]
+struct TestVec {
+    #[serde(with = "unsigned_message::json")]
+    Unsigned: UnsignedMessage,
+    Cid: String,
+    CidHexBytes: String,
+    PrivateKey: String,
+    #[serde(with = "signature::json")]
+    Signature: Signature,
+}
+
+#[test]
+fn signing_test() {
+    let mut file = File::open("../serialization-vectors/message_signing.json").unwrap();
+    let mut string = String::new();
+    file.read_to_string(&mut string).unwrap();
+
+    let vectors: Vec<TestVec> =
+        serde_json::from_str(&string).expect("Test vector deserialization failed");
+    for test_vec in vectors {
+        let test = base64::decode(test_vec.PrivateKey).unwrap();
+        let priv_key = PrivateKey::from_bytes(&test).unwrap();
```

Comment:
can you add a todo for setting up a private key based on the signature type? this is assuming this is always going to be bls signatures